File: LeetCode/py/1742.py
#Link: https://leetcode.com/problems/maximum-number-of-balls-in-a-box/submissions/

class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        counts = [0]*46
        for i in range(lowLimit, highLimit+1):
            _sum = 0
            while i != 0:
                _sum += i % 10
                i //= 10
            counts[_sum] += 1
        return max(counts)
        

File: LeetCode/cpp/1742.cpp
//Link: https://leetcode.com/problems/maximum-number-of-balls-in-a-box/

class Solution {
public:
    int countBalls(int lowLimit, int highLimit) {
        int counts[46] = {0}, sum, max_count = 0;
        for(int i = lowLimit ; i <= highLimit ; i++) {
            sum = 0;
            for(auto x : to_string(i))
                sum += x-48;
            counts[sum]++;
            max_count = max(max_count,counts[sum]);
        }
        return max_count;
    }
};
